### PR TITLE
[chore] Upgrade excalibur and sample

### PR DESCRIPTION
**Pull request by eonarheim**
This PR upgrades the old shoot-em-up example to use modern excalibur and patterns.

TODOS:
- [x] Add scene based level instead of having a bunch of code in main
- [x] Fix game over screen

### CODE REVIEW

**Comment by code0wl** [comment]
abandoned? I am glad to help review if that is the case :)
Currently writing a book using your framework. Do not mind contributing to the OS project

**Comment by eonarheim** [comment]
@code0wl Not abandoned :) just a busy time of year for life events.

Excellent, I'd love to hear about the book! And definitely appreciate any contributions!

Perhaps we could schedule some time to chat over a video call? /cc @kamranayub @jedeen

**Comment by code0wl** [comment]
@eonarheim thanks for the clarification. Would love to have a chat with you and your team. I think we can bring some new users to the framework by working together. Can we perhaps setup a community slack for this?

File: src/resources.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import * as ex from "excalibur";
```
Comment (code0wl) [inline src/resources.ts]: Really love this idea!

File: src/actors/AnimationManager.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import * as ex from "excalibur";
```
Comment (code0wl) [inline src/actors/AnimationManager.ts]: Love this as well. Having a AnimationManager that keeps records of which animations are allowed to be played

File: src/actors/AnimationManager.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            anim: animation,
+            pos: pos.clone()
+        });
+    }
+
+    onPostUpdate() {
+        this.animations = this.animations.filter(a => !a.anim.isDone());
+    }
+
+    onPostDraw(ctx: CanvasRenderingContext2D, delta: number) {
```
Comment (code0wl) [inline src/actors/AnimationManager.ts]: Can omit delta?

**Comment by code0wl** [comment]
Reviewed as best as I could. Really nice work! :+1:

**Comment by eonarheim** [comment]
@code0wl Thanks for the review, I'll update accordingly

**Pull request merged**